akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala

```diff
@@ -24,6 +24,7 @@ final case class SnapshotMetadata(persistenceId: String, sequenceNr: Long, times
  */
 @SerialVersionUID(1L)
 final case class SaveSnapshotSuccess(metadata: SnapshotMetadata)
+  extends SnapshotProtocol.Response // Internal snapshot acknowledgement.
```

Comment:
well, the original intent was
https://github.com/akka/akka/blob/master/akka-persistence/src/main/scala/akka/persistence/Protocol.scala
```
  /**
   * Helps persistence plugin developers to differentiate
   * internal persistence extension messages from their custom plugin messages.
   */
```

1) re "These messages are user API, so not internal" - they are both public api and internal, look:
https://github.com/akka/akka/blob/master/akka-persistence/src/main/scala/akka/persistence/snapshot/SnapshotStore.scala#L32

2) re:  "Response is the best type here" - it probably better to separate internal from public api completely,
despite seeming message duplication

3) re: "What about SnapshotOffer," - it is used as external only, no internal api function,

please confirm what you want to do with this